velox/exec/Task.cpp

```diff
@@ -1550,6 +1556,172 @@ void Task::noMoreSplits(const core::PlanNodeId& planNodeId) {
   }
 }
 
+ContinueFuture Task::requestBarrier() {
+  VELOX_CHECK(supportBarrier_, "Task doesn't support barrier");
+  return startBarrier("Task::requestBarrier");
+}
+
+ContinueFuture Task::startBarrier(std::string_view comment) {
```

Comment:
Wondering, how to we handle barrier for operators / drivers that are "blocked" and are waiting on some future?

Reply:
It follows the normal execution (resume when async event finishes) and if the leaf operator is waiting for split and it will resume because of the barrier split insertion.